### PR TITLE
FrontendDialectTransformer: Don't crash when importing Cast with saturate attribute

### PR DESCRIPTION
**Pull request by mgehre-amd**
While investigating why some test case don't have error messages, I found a fix for this crash.

The Cast op got a new attribute `saturate`, but the ONNX to onnx-mlir parse assumed that all attributes on the Cast op are data types (and not a boolean like for saturate).

### CODE REVIEW

File: src/Builder/FrontendDialectTransformer.cpp [hunk 1]
```diff
@@ -868,10 +868,14 @@ class FrontendGenImpl {
     std::vector<NamedAttribute> attributes;
     for (int i = 0; i < node.attribute_size(); ++i) {
       auto attr = node.attribute(i);
-      auto mlir_type = convertONNXTypeToMLIRType(
-          builder_, static_cast<onnx::TensorProto_DataType>(attr.i()));
-      Attribute mlirAttr = TypeAttr::get(mlir_type);
-      attributes.push_back(builder_.getNamedAttr(attr.name(), mlirAttr));
+      if (attr.name() == "to") {
```
Comment (TinaAMD) [inline src/Builder/FrontendDialectTransformer.cpp:871]: Can you add the comment similar to the PR description here?

File: test/mlir/onnx/parse/cast_with_saturate.onnxtext [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+   producer_name: "backend-test"
+>
+test_cast_no_saturate_FLOAT16_to_FLOAT8E4M3FNUZ (float16[3,4] input) => (float8e4m3fnuz[3,4] output) {
+   output = Cast <saturate = 0, to = 18> (input)
+}
+// CHECK-LABEL:  func.func @main_graph
+// CHECK-SAME:   ([[PARAM_0_:%.+]]: tensor<3x4xf16> {onnx.name = "input"}) -> (tensor<3x4xf8E4M3FNUZ> {onnx.name = "output"}) {
+// CHECK:           [[VAR_0_:%.+]] = "onnx.Cast"([[PARAM_0_]]) {saturate = 0 : si64, to = f8E4M3FNUZ} : (tensor<3x4xf16>) -> tensor<3x4xf8E4M3FNUZ>
+// CHECK:           onnx.Return [[VAR_0_]] : tensor<3x4xf8E4M3FNUZ>
+// CHECK:         }
```
Comment (TinaAMD) [inline test/mlir/onnx/parse/cast_with_saturate.onnxtext:15]:
```suggestion
```

File: test/mlir/onnx/parse/cast_with_saturate.onnxtext [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+<
+   ir_version: 9,
+   opset_import: ["" : 19],
+   producer_name: "backend-test"
+>
+test_cast_no_saturate_FLOAT16_to_FLOAT8E4M3FNUZ (float16[3,4] input) => (float8e4m3fnuz[3,4] output) {
+   output = Cast <saturate = 0, to = 18> (input)
+}
+// CHECK-LABEL:  func.func @main_graph
+// CHECK-SAME:   ([[PARAM_0_:%.+]]: tensor<3x4xf16> {onnx.name = "input"}) -> (tensor<3x4xf8E4M3FNUZ> {onnx.name = "output"}) {
```
Comment (TinaAMD) [inline test/mlir/onnx/parse/cast_with_saturate.onnxtext:12]:
```suggestion
// CHECK-SAME:   ([[PARAM_0_:%.+]]: tensor<3x4xf16> {onnx.name = "input"}) -> (tensor<3x4xf8E4M3FNUZ> {onnx.name = "output"})
```

**Review by TinaAMD — APPROVED** [review]
LGTM

**Review by ttjost — APPROVED** [review]
Now I get what the other PR was about. Thanks!

**Pull request merged**